apiserver/facades/client/controller/controller.go

```diff
@@ -544,40 +488,25 @@ func (c *ControllerAPI) HostedModelConfigs(ctx context.Context) (params.HostedMo
 		return result, errors.Trace(err)
 	}
 
-	modelUUIDs, err := c.state.AllModelUUIDs()
+	hostedModels, err := c.modelService.HostedModels(ctx)
 	if err != nil {
-		return result, errors.Trace(err)
+		return result, fmt.Errorf("getting hosted models: %w", err)
 	}
 
-	for _, modelUUID := range modelUUIDs {
-		if modelUUID == c.state.ControllerModelUUID() {
-			continue
-		}
-		st, err := c.statePool.Get(modelUUID)
-		if err != nil {
-			// This model could have been removed.
-			if errors.Is(err, errors.NotFound) {
-				continue
-			}
-			return result, errors.Trace(err)
-		}
-		defer st.Release()
-		model, err := st.Model()
-		if err != nil {
-			return result, errors.Trace(err)
-		}
-
+	for _, hostedModel := range hostedModels {
 		config := params.HostedModelConfig{
-			Name:     model.Name(),
-			OwnerTag: model.Owner().String(),
+			Name:     hostedModel.Model.Name,
+			OwnerTag: names.NewUserTag(hostedModel.Model.OwnerName).String(),
 		}
-		modelConf, err := model.Config()
+
+		modelConf, err := c.modelConfigServiceGetter(hostedModel.Model.UUID).ModelConfig(ctx)
```

Comment:
Simon are discussing this, and are tending towards not returning model configuration from this call.

This would a avoid a potential read-storm for multiple concurrent callers to this method, hitting all model DBs on large controllers.

It is likely wasted effort in any case, as few if any users are going to _need_ the configuration for every model.

A better alternative is to just return the summaries and leave the burden on the user to call `show-model`.

Reply:
So the only usage I can see of this method is in `kill-controller`. If the controller teardown times out, it tries to open an environ for each model to destroy it directly, which at this stage needs to pass in the full model config.

Happy to rethink how we do this, it doesn't seem optimal.